src/pages/SubmissionDetails.js

```diff
@@ -81,28 +81,22 @@ export const SubmissionDetails = ({ submissionId }) => {
       const runsAnnotatedWithDurations = _.map(r => _.merge(r, { duration: getDuration(r.state, r.submission_date, r.last_modified_timestamp, isRunInTerminalState) }), runs)
       setRunsData(runsAnnotatedWithDurations)
 
-      // only refresh if there are Runs in non-terminal state
-      if (_.some(({ state }) => !isRunInTerminalState(state), runs)) {
-        scheduledRefresh.current = setTimeout(refresh, AutoRefreshInterval)
+      const loadedRunSetData = await loadRunSetData(signal)
+      setRunSetData(loadedRunSetData)
+
+      // only refresh if there are Run Sets/Runs in non-terminal state
+      if (_.some(({ state }) => !isRunSetInTerminalState(state), loadedRunSetData)) {
+        // Nesting these checks to avoid a "double refresh"
+        if (_.some(({ state }) => !isRunInTerminalState(state), runsAnnotatedWithDurations)) {
+          scheduledRefresh.current = setTimeout(refresh, AutoRefreshInterval)
+        }
       }
```

Comment:
This can be consolidated to checking for non-terminal state for Run Set only. This way we refresh until Run Set reaches terminal state. Otherwise in this conditional logic it could be possible that all Runs are in terminal state but the Run Set isn't and as a result we don't refresh the page at all.